src/main/java/org/verdictdb/connection/DbmsConnection.java

```diff
@@ -60,4 +64,32 @@ public DbmsQueryResult execute(SqlConvertible query) throws VerdictDBException {
   public abstract void abort();
 
   public abstract DbmsConnection copy() throws VerdictDBDbmsException;
+
+  /**
+   * @return a list of column names of primary key columns. (0-indexed)
+   */
+  public List<String> getPrimaryKey(String schema, String table) throws VerdictDBDbmsException {
+    List<Integer> primaryKeyIndexList = new ArrayList<>();
+    List<String> primaryKeyColumnName = new ArrayList<>();
+    SqlSyntax syntax = getSyntax();
+    if (syntax.getPrimaryKey(schema, table)!=null) {
+      DbmsQueryResult result = execute(syntax.getPrimaryKey(schema, table));
+      if (syntax instanceof MysqlSyntax) {
```

Comment:
I don't think we need this syntax check because "getPrimaryKey() != null" is already used.